plugins/inventory/lxd.py

```diff
@@ -666,9 +666,11 @@ def build_inventory_hosts(self):
             # add network informations
             self.build_inventory_network(instance_name)
             # add os
-            self.inventory.set_variable(instance_name, 'ansible_lxd_os', self._get_data_entry('inventory/{0}/os'.format(instance_name)).lower())
+            if self._get_data_entry('inventory/{0}/os'.format(instance_name))
+                self.inventory.set_variable(instance_name, 'ansible_lxd_os', self._get_data_entry('inventory/{0}/os'.format(instance_name)).lower())
```

Comment:
Instead of computing the value twice, better cache it in a local variable:
```suggestion
            v = self._get_data_entry('inventory/{0}/os'.format(instance_name))
            if v:
                self.inventory.set_variable(instance_name, 'ansible_lxd_os', v.lower())
```